evasquez26/angular4-startup
# S4-ONE-125-126 Request & Accept/Reject Role: Guardian

## Story / Task

see [ONE-125](https://bixlabs.atlassian.net/browse/ONE-125) & [ONE-126](https://bixlabs.atlassian.net/browse/ONE-126)


## What this PR does?

Adds functionality to request a guardian role

### Before 

You could not request

### After

You can request to be a user's guardian from their profile page

### How to test / reproduce

- Login to the app
- Go to My profile
- See that you can not request to be your own guardian
- Go to another user's profile (take a user id and go to /user/USER_ID/profile)
- See that you can request to be a guardian
- Click Become a Guardian
- Fill the information and click request role
- See a label that says you already requested a role guardian.
- Logout and login as the user you requested to be guardian of
- Go to My Profile -> Guardian requests (new)
- See a list of requests type guardian
- Click Accept and Confirm. 
- Logout and login as the requester user
- Go to the requested user's profile
- You should see that you are already a guardian
- Go to a different user and request to become a guardian
- Login as the requested user and reject the role request. You can add a comment.
- Login as the requester and go to the requested user's profile
- You should see that you can request the role again.

### TODOs
- [ ] Rebase after merging #34 

### Related PRs
N/A

## Migrations
 NO


Thank you for your time
